tests/unit/TypeResolverTest.php

```diff
@@ -407,9 +407,9 @@ public function testResolvingNullableCompoundTypes(): void
     {
         $fixture = new TypeResolver();
 
+        $this->expectException('RuntimeException');
+        $this->expectExceptionMessage('Unexpected type separator');
         $resolvedType = $fixture->resolve('?string|null|?boolean');
-
-        $this->assertSame('?string|null|?bool', (string) $resolvedType);
```

Comment:
I'm not sure if this behavior should change. I do understand that PHP doesn't allow this format and that it looks more than ugly. But this case has been added because we found some repositories using this notation. 

As this library is used to parse types in docblocks and native PHP, I think we should keep this format supported. 
It could result in a deprecation that we will remove in the future. But for now, this is a backward compatibility break which we cannot do with the number of consuming projects for this library. 

Do you think it is possible to revert this change?

Reply:
Thanks for the feedback! I did not fully appreciate the need for backwards compatibility, so I have restored support for shorthand nullable syntax in compound types as requested. Please let me know if this solution is not satisfactory.